src/Lachain.Networking/Hub/HubConnector.cs

```diff
@@ -100,6 +100,26 @@ public void Send(byte[] publicKey, byte[] message)
             CommunicationHub.Net.Hub.Send(publicKey, CompressUtils.DeflateCompress(message).ToArray());
         }
 
+        [MethodImpl(MethodImplOptions.Synchronized)]
```

Comment:
I think it is better to use separate lock for SendVal method, because Send method can be spammed and they have same lock, so Send method can prevent calling SendVal.